### DISCUSSION TITLE
Not connecting to Modbus

### DISCUSSION THREAD

**Discussion by ejbatts** [WillCodeForCats/solaredge-modbus-multi#294] [General]
**Describe the bug**
Wont connect to Modbus

**Expected behavior**
Normal connection to Modbus

**Screenshots**
![image](https://user-images.githubusercontent.com/28001698/232961074-4e9aae8d-7610-4bce-b67f-d24f319d4819.png)

**Logs**
No logs in "custom_components.solaredge_modbus" or "modbus_multi"

**Debug Logs**
Debug logs are not always necessary, but may be required for more complex issues. You can enable debug logs by adding the following to your `configuration.yaml` file and restarting Home Assistant.
```
logger:
    default: warning
    logs:
        custom_components.solaredge_modbus_multi: debug
```
Debug logging will generate a large amount of data: it is recommended to configure it, collect the data needed, then remove debug logging during normal operation.

**Home Assistant (please complete the following information):**
 - Home Assistant Core Version: 2023.4.5
 - solaredge-modbus-multi Version: 2.2.13

**Additional context**
Problem may have commenced during upgrade to HA OS 10.0 and or Core 2023.4.5
I have redownloaded Integration, I have checked inverter configurations.

**Comment by ejbatts** [comment]
In desperation I changed the Modbus/TCP port on the leader inverter to **1503** and deleted/reinstalled "solaredge-modbus-multi", and it worked!

Q. Why was the original config with port **1502** not connecting, or why did it suddenly stop connecting?

**Comment by WillCodeForCats** [comment]
Delete solaredge-modbus if you're using solaredge-modbus-multi

**Reply by ejbatts** [reply]
Yeah sure, I only added it to test the connection after the failure. Any reason for 1502 to stop working suddenly, something in HA core or OS perhaps?

**Reply by WillCodeForCats** [reply]
The logs show a bad setup with a conflicting integration, that's all I can determine from that.

**Comment by emanuelerosato** [comment]
Yesterday I configured the following integration in order to manage my system with meter. Everything worked great. Now same problem written in this post.

I have no other integrations and I haven't changed anything.

What can I do?

`Connection to (192.168.178.51, 1502) failed: timed out
22:43:51 – (ERRORE) /usr/local/lib/python3.10/site-packages/pymodbus/client/tcp.py - il messaggio si è verificato per la prima volta alle 22:43:42 e compare 2 volte
Polling frequency < 10, requiring keep modbus open.
22:43:48 – (AVVISO) SolarEdge Modbus Multi Device (integrazione personalizzata) - il messaggio si è verificato per la prima volta alle 22:43:39 e compare 2 volte
Config entry 'SolarEdge' for solaredge_modbus_multi integration not ready yet: Could not open Modbus/TCP connection to 192.168.178.51; Retrying in background
22:43:43 – (AVVISO) config_entries.py`

**Reply by emanuelerosato** [reply]
I also tried to change port passing on 1503 but the result does not change

**Reply by ejbatts** [reply]
Confirm you changed the port inside the inverter as well?

**Reply by emanuelerosato** [reply]
Yes. confirm. 
![photo_5913521766288965459_y](https://user-images.githubusercontent.com/100445998/233563803-d4b45d20-e459-49d6-b7f0-6ef08ef2b4dd.jpg)
![photo_5913521766288965460_y](https://user-images.githubusercontent.com/100445998/233563811-29150f66-a607-4fe0-95c1-545d887091d9.jpg)

**Reply by ejbatts** [reply]
Your image appears to be a follower inverter, ModbusTCP Port 1503 should only be enabled on the leader with ID of 1. The follower inverters should have sequential ID's. I have MODbus TCP disabled on them. 

Also, RS485-1 setting should be either 'leader' or 'follower' as appropriate, NOT Modbus (multi).

This information is clearly provided in the setup instructions for the integration.

**Reply by emanuelerosato** [reply]
What does this mean? I did the configuration, did I do something wrong?

I have both inverter and meter.

One thing I've found, if I ping the inverter's ip or a nc on ip and port 1503, it doesn't give me any results.

**Reply by ejbatts** [reply]
I edited, read my post again, does it make sense?
